src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -279,11 +355,7 @@ private static IReadOnlyList<object> AsReadOnlyList(IList<object> metadata)
         // return null for plain RequestDelegates that have not been modified by filters so we can just pass back the original RequestDelegate.
         if (filterPipeline is null && factoryContext.Handler is RequestDelegate)
         {
-            // Make sure we're still not handling a return value.
-            if (!returnType.IsGenericType || returnType.GetGenericTypeDefinition() != typeof(Task<>))
-            {
-                return null;
-            }
+            return null;
```

Comment:
Since we [stopped passing normal RequestDelegates to RDF just because it returned a `Task<T>`](https://github.com/dotnet/aspnetcore/pull/42519/files#diff-f8807c470bcc3a077fb176668a46df57b4bb99c992b6b7b375665f8bf3903c94L202-L207), we no longer have to check for that case here and can preserve the original RequestDelegate any time the filter pipeline returns the original `next` invocation.